academy: minor restructure (#431)
apify/apify-docs: content/academy/web_scraping_for_beginners/challenge/initializing_and_setting_up.md

---
title: Initializing & setting up
description: When you collect links from a web page, you often end up with a lot of irrelevant URLs. Learn how to filter the links to only keep the ones you need.
menuWeight: 1
paths:
- web-scraping-for-beginners/challenge/initializing-and-setting-up
---

# Initialization & setting up

The Crawlee CLI makes it extremely easy for us to set up a project in Crawlee and hit the ground running. Navigate to the directory you'd like your project's folder to live, then open up a terminal instance and run the following command:

```shell

> npx crawlee create demo-actor
> ```
> 
> > You don't have to call it **demo-actor**, but that's what we'll be calling it in this tutorial.

Reviewer: I wouldn't call it demo-**actor**. It's the open-source part of the course so actors are irrelevant here.
```suggestion
npx crawlee create amazon-crawler
```

> You don't have to call it **amazon-crawler**, but that's what we'll be calling it in this tutorial.
```

Once you run this command, you'll get prompted into a menu which you can navigate using your arrow keys. Each of these options will generate different boilerplate code when selected. We're going to work with CheerioCrawler today, so we'll select the **CheerioCrawler template project** template, then press **Enter**.

![Crawlee CLI "create" command]({{@asset web_scraping_for_beginners/challenge/images/crawlee-create.webp}})

> Once it's completed, open up the **demo-actor** folder that was generated by the `npx crawlee create` command. We're going to modify the **main.js** boilerplate to fit our needs:

Reviewer: ```suggestion
Once it's completed, open up the **amazon-crawler** folder that was generated by the `npx crawlee create` command. We're going to modify the **main.js** boilerplate to fit our needs:
```

```JavaScript
// main.js
import { CheerioCrawler, KeyValueStore, log } from 'crawlee';
import { router } from './routes.js';

// Grab our keyword from the input

> const { keyword = 'iphone' } = (await KeyValueStore.getInput()) ?? {};

Reviewer: Let's fail fast here. The original code would be super confusing for beginners.

```suggestion
const { keyword } = await KeyValueStore.getInput();
```

const crawler = new CheerioCrawler({
    requestHandler: router,
});

> // Add our initial requests
> await crawler.addRequests([
>     {
>         // Turn the inputted keyword into a link we can make a request with
>         url: `https://www.amazon.com/s/ref=nb_sb_noss?url=search-alias%3Daps&field-keywords=${keyword}`,
>         label: 'START',
>         userData: {
>             keyword,
>         },
>     },
> ]);

Reviewer: ```suggestion
```

log.info('Starting the crawl.');

> await crawler.run();

Reviewer: ```suggestion
await crawler.run([{
    // Turn the keyword into a link we can make a request with
    url: `https://www.amazon.com/s/ref=nb_sb_noss?url=search-alias%3Daps&field-keywords=${keyword}`,
    label: 'START',
    userData: {
        keyword,
    },    
}]);
```

log.info('Crawl finished.');
```

```JavaScript
// routes.js
import { createCheerioRouter } from 'crawlee';

export const router = createCheerioRouter();

router.addDefaultHandler(({ log }) => {
    log.info('Route reached.')
});
```

> Finally, we'll modify our input file in **storage/key_value_stores/default/INPUT.json** to look like this:

Reviewer: Crawlee now supports `INPUT.json` in project root.

```suggestion
Finally, we'll add the following input file to **INPUT.json** in the project's root directory (next to `package.json`, `node_modules` and others)
```
